core/controllers/acl_decorators.py

```diff
@@ -2753,6 +2753,73 @@ def test_can_accept_suggestion(
     return generate_decorator_for_handler
 
 
+def can_update_suggestions(handler):
+    """Function that generates a decorator for a given handler.
+
+    Args:
+        handler: function. The function to be decorated.
+
+    Returns:
+        function. The newly decorated function that has common checks and
+        permissions specified by passed in decorator.
+
+    Raises:
+        NotLoggedInException. The user is not logged in.
+        UnauthorizedUserException. The user does not have credentials to
+            edit this suggestion.
+    """
+    def test_can_update_suggestion(
+            self, suggestion_id, **kwargs):
+        """Returns a handler to test whether a
+        suggestion can be updated based on the user roles.
+
+        Args:
+            suggestion_id: str. The suggestion id.
+            **kwargs: *. Keyword arguments.
+
+        Returns:
+            function. The handler for updating a
+            suggestion.
+
+        Raises:
+            NotLoggedInException. The user is not logged in.
+        """
+        if not self.user_id:
+            raise base.UserFacingExceptions.NotLoggedInException
+        user_actions = user_services.get_user_actions_info(
+            self.user_id
+        ).actions
+        if role_services.ACTION_ACCEPT_ANY_SUGGESTION in user_actions:
+            return handler(self, suggestion_id, **kwargs)
+
+        if len(suggestion_id.split('.')) != 3:
+            raise self.InvalidInputException(
+                'Invalid format for suggestion_id.'
+                ' It must contain 3 parts separated by \'.\'')
+
+        suggestion = suggestion_services.get_suggestion_by_id(suggestion_id)
+
+        if suggestion is None:
+            raise self.PageNotFoundException
+
+        if suggestion.suggestion_type == (
+                feconf.SUGGESTION_TYPE_TRANSLATE_CONTENT):
+            if user_services.can_review_translation_suggestions(
+                    self.user_id,
+                    language_code=suggestion.change.language_code):
+                return handler(self, suggestion_id, **kwargs)
+
+        if suggestion_services.check_can_resubmit_suggestion(
+                suggestion_id, self.user_id):
+            return handler(self, suggestion_id, **kwargs)
+        else:
+            raise base.UserFacingExceptions.UnauthorizedUserException(
+                'You do not have credentials to resubmit this suggestion.')
```

Comment:
This should be updated to tell the user they can't update the suggestion.

Reply:
Done